kubetools/cli/show.py

```diff
@@ -98,6 +99,17 @@ def _get_command(item):
     return get_object_annotations_dict(item).get('description')
 
 
+def _get_cronjob_status(item):
+    # CronJob didn't start yet
+    if item.status.last_schedule_time is None and item.status.last_successful_time is None:
+        return '0/1'
+    # CronJob is running
+    elif item.status.last_schedule_time is not None:
+        return '1/1'
+    else:
+        return 'unknown'
```

Comment:
I've changed the `_get_cronjob_status` function

Please let me know what you think ?

Reply:
I'm not clear yet how this works, or even what this code is used for. Have you tested it? What command did you use to test?

Reply:
Here is the code that I added:
```
def _get_cronjob_status(item):
    # CronJob is running
    if item.status.active is not None:
        return '1/1'
    else:
        # CronJob didn't start yet
        if item.status.last_schedule_time is None:
            return '0/1'
        # CronJob is running but the Job didn't start yet
        else:
            return '1/1'
```

From what I've seen, the `active` parameter in not `None` if there is a job (of this Cronjob) currently running.
So for example if you run the command: `kubetools deploy cronjob-tests`
and then if you run the command `kubetools show cronjob-test`

On the `Ready` column of the Cronjob line you will see `1/1` or `0/1` (this is the part that this function is changing, you can change the function to see how it works)

One way to see how the `Ready` column changes, it's to add a print between line `209` and `210`:
```
def _get_cronjob_status(item):
    print(f"{item.status}")
    # CronJob is running
```

So to explain the code of the function `_get_cronjob_status`:
```
if  item.status.active is not None --> Job of Cronjob is currently running
else:
    if item.status.last_schedule_time is None --> 0 Job has been ran yet, Cronjob is not ready
    else --> there is a job of Cronjob that has ran and finished, so waiting for next job to start (Cronjob is running but doesn't appear active, because there isn't a Job of this Cronjob running at this moment)
```

Reply:
OK, makes more sense now. Then I'd change the output to be:
```
if  item.status.active is not None:
    Job of Cronjob is currently running
    -> "0/1"
elif item.status.last_schedule_time is None:
    0 Job has been ran yet, Cronjob is not ready
    -> "0/0"
else:
    there is a job of Cronjob that has ran and finished, so waiting for next job to start (Cronjob is running but doesn't appear active, because there isn't a Job of this Cronjob running at this moment)
    -> "1/1"

Reply:
One question

Why do we put  `-> "0/1"` if active  `is not None` ?
If active is not None that means that the Cronjob is up and there is a job running. It should be `-> "1/1"`

And is the `else:` part, it should be `-> "0/1"`
Because that means that the Cronjob is up but there is no job running at this moment.

I think it should be like below no ?

```
if  item.status.active is not None:
    Job of Cronjob is currently running
    -> "1/1"
elif item.status.last_schedule_time is None:
    0 Job has been ran yet, Cronjob is not ready
    -> "0/0"
else:
    there is a job of Cronjob that has ran and finished, so waiting for next job to start (Cronjob is running but doesn't appear active, because there isn't a Job of this Cronjob running at this moment)
    -> "0/1"
```

Reply:
My idea is that the first digit is for the success of a job, while the second one is for the success of the job scheduling. Actually, I think I missed something and it should be:
```
if  item.status.active is not None:
    Job is currently running (implies successfully started)
    -> "?/1"
elif item.status.last_successful_time is not None:
   Job has been successful (implies successfully started)
    -> "1/1"
elif item.status.last_schedule_time is not None:
    Job has been scheduled
    -> "0/1"
else:
    Job has never been scheduled (error in CronJob?)
    -> "0/0"
```

Maybe using a word describing each situation would help.

Also, the [k8s documentation](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) is not giving anything about the dependencies between the statuses of a CronJob. So I'm making the assumptions that:
* a Job cannot be running if it's not been scheduled
* a Job cannot be successful if it's not been scheduled
* a Job cannot be scheduled if the CronJob was not created in k8s


Let me know what you think, I might be missing something

Reply:
Okay make sense! This looks good, I did the change

Here is the commit: https://github.com/EDITD/kubetools/pull/118/commits/1450de2fea4ebaa062bb98292aabd67e8ff500ad